File: Exams/E1/python/e1.py
class Student:
    Gr = []
    We = []
    def init(self,Gr,We):
        for i in range(len(Gr)):
            self.Gr.append(Gr[i])
            self.We.append(We[i])
    

    def Avg(self):
        sum = 0
        a = 0
        for i in range(len(st.Gr)):
        
            sum+=self.Gr[i]*self.We[i]
            a+=self.We[i]
        
        return sum/a

if __name__ == "__main__":
    Gr = [20,16,18,13]
    We = [3,2,2,1]
    st = Student()
    print(st.Avg())

File: Exams/E1/cpp/E1.cpp
#include <iostream>
using namespace std;
class Student
    {
        public :
        int sum=0;
        int G[100];
        int W[100];
        void grades(int i,int q)
        {
            G[i] = q;
        }
        void weights(int i,int d)
        {
            W[i] = d;
        }
        int Avg(int sizee, int a)
        {
            for(int i=0; i<sizee; i++)
            {
                sum += G[i]*W[i];
            }
            return sum/a;
        }
    };
int main()
{
    int a=0;
    int k=0;
    Student st;
    int b;
    while(cin.get() != '\n')
    {
        cin >> b;
        st.grades(k,b);
        k++;
    }
    int i=0;
    int c;
    while(cin.get() != '\n')
    {
        cin >> c;
        a += c;
        st.weights(i,c);
        i++;
    }
    cout << st.Avg(k,a);
    return 0;
}
